classes/migration/upgrade/v3_4_0/I8866_DispatchRegionCodesFixingJobs.php

```diff
@@ -58,35 +73,45 @@ public function up(): void
             $jobs = [];
             foreach (array_keys($mappings) as $country) {
-                $jobs[] = new FixRegionCodes($country);
+                $jobs[] = new RegionMappingTmpInsert($country);
+                for ($i = 0; $i < $geoDailyChunksNo; $i++) {
+                    $startId = ($i * $chunkSize) + 1;
+                    $endId = min(($i + 1) * $chunkSize, $geoDailyIdMax);
+                    $jobs[] = new PreFixRegionCodesDaily($startId, $endId);
+                }
+                for ($i = 0; $i < $geoMonthlyChunksNo; $i++) {
+                    $startId = ($i * $chunkSize) + 1;
+                    $endId = min(($i + 1) * $chunkSize, $geoMonthlyIdMax);
+                    $jobs[] = new PreFixRegionCodesMonthly($startId, $endId);
+                }
+                for ($i = 0; $i < $geoDailyChunksNo; $i++) {
+                    $startId = ($i * $chunkSize) + 1;
+                    $endId = min(($i + 1) * $chunkSize, $geoDailyIdMax);
+                    $jobs[] = new FixRegionCodesDaily($startId, $endId);
+                }
+                for ($i = 0; $i < $geoMonthlyChunksNo; $i++) {
+                    $startId = ($i * $chunkSize) + 1;
+                    $endId = min(($i + 1) * $chunkSize, $geoMonthlyIdMax);
+                    $jobs[] = new FixRegionCodesMonthly($startId, $endId);
+                }
+                $jobs[] = new RegionMappingTmpClear();
             }
-
-            Bus::batch($jobs)
-                ->then(function (Batch $batch) {
-                    // drop the temporary index
-                    Schema::table('metrics_submission_geo_daily', function (Blueprint $table) {
-                        $sm = Schema::getConnection()->getDoctrineSchemaManager();
-                        $indexesFound = $sm->listTableIndexes('metrics_submission_geo_daily');
-                        if (array_key_exists('metrics_submission_geo_daily_tmp_index', $indexesFound)) {
-                            $table->dropIndex(['tmp']);
-                        }
-                    });
-                    Schema::table('metrics_submission_geo_monthly', function (Blueprint $table) {
-                        $sm = Schema::getConnection()->getDoctrineSchemaManager();
-                        $indexesFound = $sm->listTableIndexes('metrics_submission_geo_monthly');
-                        if (array_key_exists('metrics_submission_geo_monthly_tmp_index', $indexesFound)) {
-                            $table->dropIndex(['tmp']);
-                        }
-                    });
-
-                    // drop the temporary table
-                    if (Schema::hasTable('region_mapping_tmp')) {
-                        Schema::drop('region_mapping_tmp');
-                    }
+            $jobs[] = new CleanTmpChangesForRegionCodesFixes();
+            Bus::chain($jobs)
+                ->catch(function (Throwable $e) {
```

Comment:
There should be some error logging, the `Throwable` covers even syntax errors at included files.